src/main/java/org/assertj/core/api/AbstractAssert.java

```diff
@@ -50,6 +50,8 @@
  */
 public abstract class AbstractAssert<SELF extends AbstractAssert<SELF, ACTUAL>, ACTUAL> implements Assert<SELF, ACTUAL> {
 
+  public static boolean throwUnsupportedExceptionOnEquals = true;
```

Comment:
add a comment with a link to https://github.com/joel-costigliola/assertj-core/issues/1128 to make it clear why this field exists.